aesd-char-driver/main.c

```diff
@@ -175,12 +187,143 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
+    struct aesd_buffer_entry *cur_entry;
+    loff_t f_pos_off = 0; 
+    size_t entry_off = 0;
+    struct aesd_seekto seekto;
+
+    PDEBUG("ioctl cmd %u with arg %lld", cmd, arg);
+
+    // Ensure valid command
+    if(_IOC_TYPE(cmd) != AESD_IOC_MAGIC)
+        return -ENOTTY;
+    if(_IOC_NR(cmd) > AESDCHAR_IOC_MAXNR)
+        return -ENOTTY;
+
+    PDEBUG("ioctl cmd passed validity checks\n");
+
+    // Integrity checks on passed data
+    if (_IOC_DIR(cmd) & _IOC_READ)
+        err = !access_ok_wrapper(VERIFY_WRITE, (void __user *)arg, _IOC_SIZE(cmd));
+    else if (_IOC_DIR(cmd) & _IOC_WRITE)
+        err =  !access_ok_wrapper(VERIFY_READ, (void __user *)arg, _IOC_SIZE(cmd));
+    if (err)
+        return -EFAULT;
+
+    PDEBUG("ioctl cmd passed integrity checks\n");
+
+    switch(cmd) {
+        case AESDCHAR_IOCSEEKTO:
+            if(copy_from_user(&seekto, (const void __user *)arg, sizeof(seekto)) != 0) {
+                return -EFAULT;
+            } else {
+                PDEBUG("ioctl ind: %u offset: %u\n", seekto.write_cmd, seekto.write_cmd_offset);
+
+                // lock cbuffer access
+                if(mutex_lock_interruptible(&dev->mut) != 0) {
+                    PDEBUG("ioctl: mut lock err!\n");
+                    return -ERESTARTSYS;
+                }
+
+                absind = (seekto.write_cmd + dev->circ_buf.out_offs) % AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
+                
+                // validate arg
+                if(seekto.write_cmd > AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED) {
+                    PDEBUG("Command index %u is out of range!\n", seekto.write_cmd);
+                    return -EFAULT;
+                } else if(dev->circ_buf.entry[absind].buffptr == NULL) {
+                    PDEBUG("Command index %u  is not yet written!\n", seekto.write_cmd);
+                    return -EFAULT;
+                } else if(dev->circ_buf.entry[absind].size <= seekto.write_cmd_offset) {
+                    PDEBUG("Command offset %u is beyond command %u size of %u\n", seekto.write_cmd_offset, seekto.write_cmd, dev->circ_buf.entry[absind].size);
+                    return -EFAULT;
+                }
+
+                // Get offset to specified seek
+                target = dev->circ_buf.entry[absind].buffptr + seekto.write_cmd_offset;
+                while(f_pos_off < SOME_REASONABLE_CUTOFF_VALUE) {
+                    cur_entry = aesd_circular_buffer_find_entry_offset_for_fpos(&dev->circ_buf, (size_t)f_pos_off, &entry_off);
+                    cur = &cur_entry->buffptr[entry_off];
+
+                    PDEBUG("{%.*s: %c} ", (int)cur_entry->size, cur_entry->buffptr, *cur);
+
+                    if(cur == target) {
+                        break;
+                    } else {
+                        f_pos_off++;
+                    }
+                }
+                PDEBUG("\n");
+
+                // update fpos
+                mutex_lock(&filp->f_pos_lock);
```

Comment:
This is a good thing to worry about, however I don't think `f_pos_lock` is going to ultimately accomplish what you are trying to accomplish here.  Let's discuss in code review.